Add java-imports-scan-local-jars

erjoalgo opened the pull request:
a function to scan the local ~/.m2 repo for class-package associations
addressess #15

dakrone commented on java-imports.el:
```diff
@@ -275,6 +275,35 @@ package and cache it for future statements."
+	  (when (string-match "\\(.*\\)/\\(.*\\)[.]class" line)
+	    (let ((class (intern (match-string 2 line)))
+		  (package-name
+		   (->>
```
This adds a dependency on the `dash.el` library, can you add that as a dependency to the file?

dakrone commented on java-imports.el:
```diff
@@ -275,6 +275,35 @@ package and cache it for future statements."
+for that class in the cache.
+This is currently a synchronous and potentially slow operation, but
+hopefully faster than adding imports manually or using eclipse"
+  (labels ((shell-command-to-lines (cmd)
```
Can you use `cl-labels` instead of `labels`

erjoalgo commented:
Thanks for reviewing. I totally forgot to remove the -> dash dependency, and I also removed the s.el dependency. I also didn't want to overwrite an existing entry which the user may have manually introduced over an auto-generated one.

dakrone commented:
Merged this, thanks a ton!

erjoalgo commented:
Thanks. It looks like replace-regexp-in-string still creates a dependency on the subr-x, while the travis test is using emacs pre-24.4. At the same I added back the s.el dependency. But even if I switched to s-replace, it is merely a proxy for replace-regexp-in-string:

(defun s-replace (old new s)
  "Replaces OLD with NEW in S."
  (replace-regexp-in-string (regexp-quote old) new s t t))

even s-split

(defun s-split (separator s &optional omit-nulls)
  "Split S into substrings bounded by matches for regexp SEPARATOR.
If OMIT-NULLS is non-nil, zero-length substrings are omitted.

This is a simple wrapper around the built-in `split-string'."
  (split-string s separator omit-nulls))

I can't remember the pre-emacs-24.4 way of doing these things, and I'm not sure if it's worth upgrading the emacs version in the test, but I'm sure users of this function would be able to figure it out.

It's interesting that the ac1f24f build passed the tests even though the `replace-regexp-in-string` was already there.
